src/main/java/com/gam/api/domain/user/service/UserServiceImpl.java

```diff
@@ -283,38 +283,35 @@ public WorkPortfolioGetResponseDTO getPortfolio(Long requestUserId, Long userId)
     public List<UserDiscoveryResponseDTO> getDiscoveryUsers(Long userId, int[] tags){
         List<UserScrapUserQueryDto> users;
 
-        if (tags.length == 0) {
+        if (tags.length == 0) { // user 정보 조회 태그가 걸린 경우와 아닌 경우
             users = userRepository.findAllDiscoveryUser(userId); //TODO - user 관련 쿼리 잡기 , 동적 쿼리 필요,,
         }
         else {
             users = userRepository.findAllDiscoveryUserWithTag(userId, tags);
         }
 
-        return users.stream().map((dto) -> {
-            val firstWorkId = dto.user().getFirstWorkId();
-            Work firstWork;
-
-            if (firstWorkId == null && !dto.user().getActiveWorks().isEmpty()) { // User 권한 에러
-                firstWork = workRepository.findFirstByUserIdAndIsActiveOrderByCreatedAtDesc(dto.user().getId(), true)
-                        .orElse(Work.builder()
-                                .user(dto.user())
-                                .photoUrl("해당하는 작업물을 찾을 수 없습니다.")
-                                .detail("해당하는 작업물을 찾을 수 없습니다.")
-                                .title("해당하는 작업물을 찾을 수 없습니다.")
-                                .build());
-                dto.user().setUserStatus(UserStatus.NOT_PERMITTED);
-            }
-            else {
-                firstWork = dto.user().getActiveWorks().stream()
-                        .filter(work -> dto.user().getFirstWorkId().equals(work.getId()))
-                        .findFirst().get();
-            }
+        Map<User, Boolean> scrapMap = new HashMap<User,Boolean>(); // user 별 스크랩 한 경우와 아닌 경우 식별을 위한 map
+        for (UserScrapUserQueryDto user:users) {
+            scrapMap.put(user.user(), user.scrapStatus());
+        }
+
+        List<Work> workAll = new LinkedList<Work>();
+
+        for (UserScrapUserQueryDto dto:users) { // 모든 work 가져오는 로직 -> 이 부분 어떻게 수정해야 할 지 고민을 해봐야 할 것 같아요
+            workAll.addAll(dto.user().getWorks());
+        }
+
+
+        workAll = workAll.stream() // 최근 수정된 날짜 기준 정리
+                .sorted(Comparator.comparing(Work::getModifiedAt).reversed())
+                .collect(Collectors.toList());
 
```

Comment:
flatMap으로 다 가져오고, stream을 적극적으로 활용하면 가독성이 좋아질거 같아요.

```suggestion
    Map<User, Boolean> scrapMap = users.stream()
            .collect(Collectors.toMap(UserScrapUserQueryDto::user, UserScrapUserQueryDto::scrapStatus));

    // 모든 work를 가져와 최근 수정된 날짜 기준으로 정리
    List<Work> workAll = users.stream()
            .flatMap(dto -> dto.user().getWorks().stream())
            .sorted(Comparator.comparing(Work::getModifiedAt).reversed())
            .collect(Collectors.toList());
```